src/library_egl.js

```diff
@@ -27,16 +49,69 @@ var LibraryEGL = {
         EGL.setErrorCode(0x3008 /* EGL_BAD_DISPLAY */);
         return 0;
       }
-      // TODO: read attribList.
       if ((!config || !config_size) && !numConfigs) {
         EGL.setErrorCode(0x300C /* EGL_BAD_PARAMETER */);
         return 0;
       }
-      if (numConfigs) {
-        {{{ makeSetValue('numConfigs', '0', '1', 'i32') }}}; // Total number of supported configs: 1.
+
+      // config variables to compare against
+      var alpha = 0;
+      var samples = 0;
+      var buffers = 0;
+      var depth = 0;
+      var stencil = 0;
+
+      if (attribList) {
+        while (1) {
+          var param = {{{ makeGetValue('attribList', '0', 'i32') }}};
+          if (param == 0x3038) { // EGL_NONE
```

Comment:
Makes sense. While we're at it, would it be a good idea to move all of the EGL constants that are sprinkled about as magic numbers into the EGL object, e.g. EGL.DEPTH_SIZE, EGL.STENCIL_SIZE? Wouldn't have to be all 170 obviously, just the subset that are used.